packages/build-tools/src/steps/functions/cache.ts

```diff
@@ -0,0 +1,122 @@
+import crypto from 'crypto';
+
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+} from '@expo/steps';
+
+function createCacheKey(cacheKey: string, paths: string[]): string {
+  const hash = crypto.createHash('sha256');
+  hash.update(paths.sort().join(''));
+
+  const pathsHash = hash.digest('hex');
+
+  return `${cacheKey}-${pathsHash}`;
```

Comment:
Should this be the case? That cache key includes paths hash? Or should the cache key be just the key that the user provided? How does Github Actions caching work?

Reply:
GIthub Actions only uses provided key, but the problem would be when user adds new paths to cache without changing key. I can't imagine use case when user doesn't want to automatically update cache file content.

Reply:
Yea, makes sense. Also, GitHub _does_ take paths into consideration, see https://github.com/actions/cache?tab=readme-ov-file#cache-version. Maybe there's something else we should take in from what they do.